x-pack/plugins/fleet/public/applications/fleet/components/fleet_server_instructions/hooks/use_fleet_server_host.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { useCallback, useState } from 'react';
+
+import { sendPutSettings, useGetSettings } from '../../../hooks';
+
+const URL_REGEX = /^(https?):\/\/[^\s$.?#].[^\s]*$/gm;
```

Comment:
I think we want to force `https` for the fleet server URL

Reply:
Does this affect environments where we pass `--insecure` to the Fleet server command? e.g. local dev?

Reply:
I think even for local dev, we are using `https` with fleet server (at least for 8.2 see https://github.com/elastic/kibana/pull/129371)  we removed the `--fleet-server-insecure-http` flag so by default Fleet server start with self signed certificate and we pass `--insecure` when we enroll agent after to not validate self signed certificate but it's still an https connection. Does it make thing clearer?

Reply:
Made some changes to enforce https URL's here and also added better error reporting for specific error cases.